src/libsyntax/parse/parser.rs

```diff
@@ -2777,9 +2792,13 @@ impl<'a> Parser<'a> {
 
                 // Parse the close delimiter.
                 let close_span = self.span;
-                self.bump();
                 self.open_braces.pop().unwrap();
 
+                // Don't bump past the EOF.
+                if self.token != token::Eof {
+                    self.bump();
```

Comment:
Perhaps a utility function like `self.bump_even_through_eof()` would be suitable?